src/NServiceBus.Core.Tests/Recoverability/SecondLevelRetries/SecondLevelRetriesTests.cs

```diff
@@ -113,12 +114,32 @@ public void ShouldDefaultRetryCountToZeroIfNoHeaderIsFound()
             Assert.AreEqual("1", fakeDispatchPipeline.DispatchContext.Get<OutgoingMessage>().Headers[Headers.Retries]);
         }
 
-        PhysicalMessageProcessingStageBehavior.Context CreateContext(string messageId, int currentRetryCount)
+        [Test]
+        public void ShouldRevertMessageBodyWhenDispatchingMessage()
```

Comment:
it is, but still much easier to reproduce the issue than in the acceptance test ;)